frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteCompletionContributor.java

```diff
@@ -0,0 +1,63 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.psi.PsiFile;
+import git4idea.repo.GitRepositoryManager;
+import io.vavr.collection.List;
+
+public class MacheteCompletionContributor extends CompletionContributor {
+
+  @Override
+  public void fillCompletionVariants(CompletionParameters parameters, CompletionResultSet result) {
+    PsiFile file = parameters.getOriginalFile();
+    var project = file.getProject();
+
+    var gitRepository = List.ofAll(GitRepositoryManager.getInstance(project).getRepositories())
+        .find(r -> file.getVirtualFile().getPath().startsWith(r.getRoot().getPath()));
+
+    if (gitRepository.isEmpty()) {
+      return;
+    }
+
+    var branchNames = List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
```

Comment:
Actually one idea, feel free to open a separate issue ofc... skip the branches that are already listed in .git/machete from completion

Reply:
And another issue (not sure if you've already started working on that): underline all non-existent branches in .git/machete with red squiggles

Reply:
issue #325 opened